lib/core/src/sync/model/data.rs

```diff
@@ -0,0 +1,194 @@
+    pub(crate) swap_id: String,
+    pub(crate) preimage: String,
+    pub(crate) create_response_json: String,
+    pub(crate) direction: Direction,
+    pub(crate) lockup_address: String,
+    pub(crate) claim_fees_sat: u64,
+    pub(crate) claim_private_key: String,
+    pub(crate) refund_private_key: String,
+    pub(crate) timeout_block_height: u32,
+    pub(crate) payer_amount_sat: u64,
+    pub(crate) receiver_amount_sat: u64,
+    pub(crate) accept_zero_conf: bool,
+    pub(crate) created_at: u32,
+    pub(crate) description: Option<String>,
+    pub(crate) claim_address: Option<String>,
+}
+
+impl ChainSyncData {
+    pub(crate) fn merge(&mut self, other: &Self, updated_fields: &[String]) {
+        for field in updated_fields {
+            match field.as_str() {
+                "description" => clone_if_changed(&mut self.description, &other.description),
+                "claim_address" => clone_if_changed(&mut self.claim_address, &other.claim_address),
+                "accept_zero_conf" => self.accept_zero_conf = other.accept_zero_conf,
+                _ => continue,
+            }
+        }
+    }
+}
+
+impl From<ChainSwap> for ChainSyncData {
+    fn from(value: ChainSwap) -> Self {
+        Self {
+            swap_id: value.id,
+            preimage: value.preimage,
+            create_response_json: value.create_response_json,
+            direction: value.direction,
+            lockup_address: value.lockup_address,
+            claim_address: value.claim_address,
+            claim_fees_sat: value.claim_fees_sat,
+            claim_private_key: value.claim_private_key,
+            refund_private_key: value.refund_private_key,
+            timeout_block_height: value.timeout_block_height,
+            payer_amount_sat: value.payer_amount_sat,
+            receiver_amount_sat: value.receiver_amount_sat,
+            accept_zero_conf: value.accept_zero_conf,
+            created_at: value.created_at,
+            description: value.description,
+        }
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub(crate) struct SendSyncData {
+    pub(crate) swap_id: String,
+    pub(crate) invoice: String,
+    pub(crate) create_response_json: String,
+    pub(crate) refund_private_key: String,
+    pub(crate) payer_amount_sat: u64,
+    pub(crate) receiver_amount_sat: u64,
+    pub(crate) created_at: u32,
+    pub(crate) preimage: Option<String>,
+    pub(crate) payment_hash: Option<String>,
+    pub(crate) description: Option<String>,
+}
+
+impl SendSyncData {
+    pub(crate) fn merge(&mut self, other: &Self, updated_fields: &[String]) {
+        for field in updated_fields {
+            match field.as_str() {
+                "preimage" => clone_if_changed(&mut self.preimage, &other.preimage),
+                "payment_hash" => clone_if_changed(&mut self.payment_hash, &other.payment_hash),
+                "description" => clone_if_changed(&mut self.description, &other.description),
```

Comment:
Would the `payment_hash` or `description` change and need merging after the swap creation?